### DISCUSSION TITLE
Can microsoft put more efforts on winget to make sure all packages are up-to-date?

### DISCUSSION THREAD

**Discussion by sgon00** [microsoft/winget-pkgs#38051] [General] [answered]
Hi, I am new to windows and come from Linux world (sometimes I use MacOS too). When windows 11 comes out, I decided to try it. Since I come from Linux world, `winget` is the right tool to me for packages installation. But when I try to install apps via `winget`, I found out many apps are not up-to-date. For instance:

```
> winget show Google.AndroidStudio

Found Android Studio [Google.AndroidStudio]
Version: 2020.3
Publisher: Google LLC
.......
......
Installer:
  Type: Nullsoft
  Locale: en-US
  Download Url: https://redirector.gvt1.com/edgedl/android/studio/install/2020.3.1.25/android-studio-2020.3.1.25-windows.exe
```

The version in winget is `android-studio-2020.3.1.25-windows.exe`, but when I check the website https://developer.android.com/studio, the latest version is `android-studio-2020.3.1.26-windows.exe`.

This is just one example. I met such issue for many apps that I tried to install.
Is `winget` a serious project in Microsoft? I am wondering if many people are using it or not. 
Should I find another replacement for windows package manager?
Why are there many outdated packages in winget repo?
Can microsoft put more efforts on winget to make sure all packages are up-to-date?

Thanks a lot.

**Comment by vedantmgoyal9** [comment, marked as answer]
It is impossible for a single man to find sources for more than 2K packages and add them to the automation, community support is highly needed. You can report packages that have some API/source where we can query the latest versions along with download URLs here: https://github.com/vedantmgoyal2009/winget-pkgs-automation/ and I will add them to the automation so that they will be kept up to date :-)

> Is `winget` a serious project in Microsoft? I am wondering if many people are using it or not.

WinGet is new in the world and hence it will take some time. Many key features like support for dependencies, portable apps, etc... are still in the development phase. The main issue is that many people don't know that the project is open source, and they can contribute to it on GitHub.

> Why are there many outdated packages in winget repo?

There are less than 10 **regular** contributors, while there are some people who contribute occasionally, so it is hard for us to maintain such a substantial number of packages up to date until they are updated by some automation or by the developer himself.

> Can microsoft put more efforts on winget to make sure all packages are up-to-date?

I have asked @denelon until when the automation will be officially approved but it's going to take a lot of time since they will be reviewing the code used thoroughly which is important.

**Reply by sgon00** [reply]
Thank you very much for the quick reply. 😊

**Reply by vedantmgoyal9** [reply]
@sgon00 I have made an edit to my comment, please give it a go.
